src/libAtomVM/opcodesswitch.h

```diff
@@ -1613,6 +1613,8 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
                             const struct Nif *nif = EXPORTED_FUNCTION_TO_NIF(func);
                             term return_value = nif->nif_ptr(ctx, arity, ctx->x);
                             if (UNLIKELY(term_is_invalid_term(return_value))) {
+                                // Restore stack base before searching for catch clause
+                                ctx->e -= (n_words + 1);
```

Comment:
As discussed offline, this is very suspicious. I suggest we investigate more before merging this.